src/bootstrap/compile.rs

```diff
@@ -198,6 +198,10 @@ pub fn rustc<'a>(build: &'a Build, target: &str, compiler: &Compiler<'a>) {
     if !build.unstable_features {
         cargo.env("CFG_DISABLE_UNSTABLE_FEATURES", "1");
     }
+    // Flag that rust llvm is in use
+    if build.config.target_config.get(target).is_none() {
```

Comment:
Looking a bit closer, there are a bunch of code paths that lazily insert a default target into the target_config HashMap but none are being hit it this particular case, e.g. the code you linked to:

```rust
let target = self.target_config.entry(self.build.clone())
.or_insert(Target::default());
```
At the point I'm performing the check, nothing has done this although it could be created if other config.mk variables that lazily insert a target are in use.

So perhaps the correct behaviour would be to also `.or_insert(Target::default())` where I need to check if `llvm_config` is set.